src/frame.rs

```diff
@@ -149,6 +149,7 @@ impl Frame {
             Frame::Error(s) => {
                 let mut bytes = Vec::with_capacity(1 + s.len() + CRLF.len());
                 bytes.push(u8::from(DataType::SimpleError));
+                bytes.extend_from_slice("ERR ".as_bytes());
```

Comment:
All errors have the prefix `ERR `

```
$ redis-cli
127.0.0.1:6379> SET incr_by_key_4 234293482390480948029348230948
OK
127.0.0.1:6379> INCRBY incr_by_key_4 10
(error) ERR value is not an integer or out of range
127.0.0.1:6379> foo
(error) ERR unknown command 'foo', with args beginning with:
127.0.0.1:6379> bar
(error) ERR unknown command 'bar', with args beginning with:
127.0.0.1:6379> set foo
(error) ERR wrong number of arguments for 'set' command
```

The create even removes the first work because it assumes the server is sending the error with that prefix.

Initially I did some string manipulation to get the tests to pass but then I realized I should just add the `ERR ` prefix and now it works.

Reply:
This is great work! I was also thinking in doing some string manipulation. Glad you found this.